Core/Src/main.c

```diff
@@ -87,14 +107,49 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  MX_TIM2_Init();
   /* USER CODE BEGIN 2 */
 
+  // Start the PWM timer
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+  HAL_TIM_PWM_Start(&htim2, TIM_CHANNEL_1);
+
+  // Pull CS Pin high to ensure no communication happens
+  HAL_GPIO_WritePin(SPI1_CSB_GPIO_Port, SPI1_CSB_Pin, GPIO_PIN_SET);
+
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+
   while (1)
   {
+	  // Pull CS Pin low to start SPI communication
+	  HAL_GPIO_WritePin(SPI1_CSB_GPIO_Port, SPI1_CSB_Pin, GPIO_PIN_RESET);
+
+	  // Transmit SPI data
+	  HAL_SPI_Transmit(&hspi1, SPI_TRANSMIT, SPI_TRANSMIT_SIZE, SPI_TIMEOUT);
+
+	  // Receive SPI data
+	  HAL_SPI_Receive(&hspi1, SPI_Recieve, SPI_RECIEVE_SIZE, SPI_TIMEOUT);
+
+	  // Pull CS Pin high to stop SPI communication
+	  HAL_GPIO_WritePin(SPI1_CSB_GPIO_Port, SPI1_CSB_Pin, GPIO_PIN_SET);
+
+	  // Store received ADC value
+	  ADC_Input = ((SPI_Recieve[1] << 8) + (SPI_Recieve[2])) * SPI_RECIEVE_BIT_MASK;
```

Comment:
For the ADC_Input calculation, could you give an explanation on why you multiply the result by the bit mask and not use the bitwise AND operator &? Also why use + instead of bitwise OR |? I see that Gordon suggested it, I will ask bec im confused lol.

Reply:
It was originally bitwise OR and AND, but I tried using the + and * operators for debugging the code as suggested by Gordon.